backend/remote-state/s3/backend_state.go

```diff
@@ -1,27 +1,112 @@
 package s3
 
 import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/hashicorp/terraform/backend"
 	"github.com/hashicorp/terraform/state"
 	"github.com/hashicorp/terraform/state/remote"
 )
 
 const (
-	keyEnvPrefix = "-env:"
+	// This will be used as directory name, the odd looking colon is simply to
+	// reduce the chance of name conflicts with existing objects.
+	keyEnvPrefix = "env:"
```

Comment:
Nope. The `-` was just in a copy&paste from the consul file. I didn't mean to have that change here, just ended up with this diff after writing the new backend_state file.